src/github.com/getlantern/flashlight/config/config.go

```diff
@@ -332,10 +338,20 @@ func (cfg Config) fetchCloudConfig() (bytes []byte, err error) {
 
 func (cfg Config) doFetchCloudConfig(proxyAddr string) ([]byte, error) {
 	log.Tracef("doFetchCloudConfig via '%s'", proxyAddr)
+
+	if proxyAddr != "" {
+		// Wait for proxy to become available
+		err := waitforserver.WaitForServer("tcp", proxyAddr, 30*time.Second)
+		if err != nil {
+			return nil, fmt.Errorf("Proxy never came up at %v: %v", proxyAddr, err)
+		}
+	}
```

Comment:
This takes care of getlantern/lantern#2274 by waiting for the proxy before trying to download via it.